benchexec/tablegenerator/__init__.py

```diff
@@ -1382,10 +1386,201 @@ def write_head_line(
+        % Arguments: benchmark name, run-set name, category, status, column name, statistic, value
+        \\providecommand\\StoreBenchExecResult[7]{\\expandafter\\newcommand\\csname#1#2#3#4#5#6\\endcsname{#7}}%
+        """
+    )
+
+    def column_statistic_to_latex_command(
+        command: LatexCommand,
+        column_statistic: ColumnStatistics,
+        command_list: List[LatexCommand],
+        **value_data,
+    ):
+        """Parses a ColumnStatistics to Latex Commands and appends them to the given command_list
+
+        The provided LatexCommand must have specified bench_name and runset_name.
+
+        Args:
+            command: LatexCommand with not empty bench_name and runset_name
+            column_statistic: ColumnStatistics to convert to LatexCommand
+            command_list: List of LatexCommands
+        """
+        if not column_statistic:
+            return
+
+        stat_value: StatValue
+        for stat_name, stat_value in column_statistic.__dict__.items():
+            if stat_value is None:
+                continue
+            column_parts = stat_name.split("_")
+            if len(column_parts) < 2:
+                column_parts.append("")
+
+            # Some colum_categories use _ in their names, that's why the column_category is the
+            # whole split list except the last word
+            command.set_command_part(
+                "column_category",
+                "".join(
+                    util.cap_first_letter(column_part)
+                    for column_part in column_parts[0:-1]
+                ),
+            ).set_command_part("column_subcategory", column_parts[-1])
+
+            for k, v in stat_value.__dict__.items():
+                # "v is None" instead of "if not v" used to allow number 0
+                if v is None:
+                    continue
+                command.set_command_part("stat_type", k if k != "sum" else "")
+                command_list.append(
+                    copy.deepcopy(command).set_command_value(v, **value_data)
+                )
+
+    latex_commands = []
+
+    for run_set, stat_list in zip(run_sets, stats):
+        current_command = LatexCommand(
+            bench_name=run_set.attributes.get("benchmarkname"),
+            runset_name=run_set.attributes.get("displayName"),  # Limiting length
+        )
+        used_column_titles = set()
+        for column, column_stats in zip(run_set.columns, stat_list):
+            column_title = (
+                column.display_title if column.display_title else column.title
+            )
+            if column_title in used_column_titles:
+                logging.warning(
+                    "Detected already used %s! "
+                    "Columns should be unique, please consider changing the name or displayTitle of this column \n"
+                    "Skipping column %s for now" % (column_title, column_title)
```

Comment:
I don't think we should silently skip columns. This will just mean that in practice users will overlook the warning and just use the commands, and they get the values of the first of these columns without knowing about it.

Either this should be a real error, or we just print identical column names, or we make the columns unique by attaching a suffix number.